runtime/protocol/http-client/jvm/test/aws/smithy/kotlin/runtime/http/interceptors/requestcompression/GzipCompressionTest.kt

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package aws.smithy.kotlin.runtime.http.interceptors.requestcompression
+
+import aws.smithy.kotlin.runtime.hashing.crc32
+import aws.smithy.kotlin.runtime.http.interceptors.decompressGzipBytes
+import aws.smithy.kotlin.runtime.io.*
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.test.runTest
+import kotlin.test.Test
+import kotlin.test.assertContentEquals
+import kotlin.test.assertEquals
+
+class GzipCompressionTest {
+    @Test
+    fun testBytesCompression() {
+        runBytesCompressionTest("")
+        runBytesCompressionTest("<Foo>bar</Foo>")
+        runBytesCompressionTest("<Baz>foo</Baz>".repeat(100))
```

Comment:
question: What is the purpose of testing two different non-zero content lengths?